src/main/java/com/redhat/cloud/common/clowder/configsource/ClowderConfigSource.java

```diff
@@ -362,43 +363,51 @@ public String getValue(String configKey) {
                         return null;
                     }
 
-                    switch (param) {
-                        case CLOWDER_ENDPOINTS_PARAM_URL:
-                            if (usesTls(endpoint)) {
-                                return "https://" + endpoint.hostname + ":" + endpoint.tlsPort;
-                            } else {
-                                return "http://" + endpoint.hostname + ":" + endpoint.port;
-                            }
-                        case CLOWDER_ENDPOINTS_PARAM_TRUST_STORE_PATH:
-                            if (usesTls(endpoint)) {
-                                ensureTlsCertPathIsPresent();
-
-                                createTruststoreFile(root.tlsCAPath);
-                                return trustStorePath;
-                            }
+                    return extractEndpointParameter(requestedEndpoint, param, endpoint);
+                } catch (IllegalStateException e) {
+                    log.errorf("Failed to load config key '%s' from the Clowder configuration: %s", configKey, e.getMessage());
+                    throw e;
+                }
+            }
 
-                            return null;
-                        case CLOWDER_ENDPOINTS_PARAM_TRUST_STORE_PASSWORD:
-                            if (usesTls(endpoint)) {
-                                ensureTlsCertPathIsPresent();
+            if (configKey.startsWith(CLOWDER_PRIVATE_ENDPOINTS)) {
+                try {
+                    if (root.privateEndpoints == null) {
```

Comment:
I feel like we could share even more code without too much trouble.

Both of these blocks of code seem to use a `List<EndpointConfig>` (we might not need to differentiate between `PrivateEndpointConfig` and `EndpointConfig`. But it can be done if the function receives a `List<? extends EndpointConfig>` (it will accept `List<EndpointConfig>` and `List<PrivateEndpointConfig>`).

The only difference I can see are the errors too, but maybe we can have some strings to pass the names e.g.
```java

processEndpoints(root.privateEndpoints, CLOWDER_PRIVATE_ENDPOINTS, "Private endpoint");

// and 

private processEndpoints(List<? extends EndpointConfig> endpoints, String clowderKey, String endpointType) {
  // ...
 final String requestedEndpointConfig = configKey.substring(clowderKey.length());
 // ..
  final String FORMAT_EXAMPLE = "[endpoint-name].[url|trust-store-path|trust-store-password|trust-store-type]";
  if (configPath.length == 1) {
                        log.warnf("%s '%s' is using the old format. Please move to the new one: %s", endpointType, requestedEndpointConfig, FORMAT_EXAMPLE);
  // etc...
  
}
```